RecoTracker/TkTrackingRegions/plugins/CandidatePointSeededTrackingRegionsProducer.h

```diff
@@ -0,0 +1,462 @@
+      m_deltaPhi_Point = regPSet.getParameter<double>("deltaPhi_Point");
+      if(m_deltaPhi_Point<0) m_deltaPhi_Point = m_deltaPhi; //For backwards compatibility
+    }
+    else if (m_seedingMode == SeedingMode::CANDIDATE_POINT_SEEDED){
+      m_deltaEta_Cand = regPSet.getParameter<double>("deltaEta_Cand");
+      m_deltaPhi_Cand = regPSet.getParameter<double>("deltaPhi_Cand");
+      m_deltaEta_Point = regPSet.getParameter<double>("deltaEta_Point");
+      m_deltaPhi_Point = regPSet.getParameter<double>("deltaPhi_Point");
+      if (m_deltaEta_Cand<0 || m_deltaPhi_Cand<0 || m_deltaEta_Point<0 || m_deltaPhi_Point<0)  throw edm::Exception(edm::errors::Configuration) << "Delta eta and phi parameters must be set separately for candidates and points in candidate+point seeding mode";
+    }   
+
+    m_precise          = regPSet.getParameter<bool>("precise");
+    m_whereToUseMeasurementTracker = RectangularEtaPhiTrackingRegion::stringToUseMeasurementTracker(regPSet.getParameter<std::string>("whereToUseMeasurementTracker"));
+    if(m_whereToUseMeasurementTracker != RectangularEtaPhiTrackingRegion::UseMeasurementTracker::kNever) {
+      token_measurementTracker = iC.consumes<MeasurementTrackerEvent>(regPSet.getParameter<edm::InputTag>("measurementTrackerName"));
+    }
+    m_searchOpt = false;
+    if (regPSet.exists("searchOpt")) m_searchOpt = regPSet.getParameter<bool>("searchOpt");
+
+    // mode-dependent z-halflength of tracking regions
+    if (m_mode == Mode::VERTICES_SIGMA)  m_nSigmaZVertex   = regPSet.getParameter<double>("nSigmaZVertex");
+    if (m_mode == Mode::VERTICES_FIXED)  m_zErrorVetex     = regPSet.getParameter<double>("zErrorVetex");
+    m_nSigmaZBeamSpot = -1.;
+    if (m_mode == Mode::BEAM_SPOT_SIGMA)
+    {
+      m_nSigmaZBeamSpot = regPSet.getParameter<double>("nSigmaZBeamSpot");
+      if (m_nSigmaZBeamSpot < 0.)
+	throw edm::Exception(edm::errors::Configuration) << "nSigmaZBeamSpot must be positive for BeamSpotSigma mode!";
+    }
+  }
+  
+  ~CandidatePointSeededTrackingRegionsProducer() override {}
+    
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+
+    desc.add<std::string>("mode", "BeamSpotFixed");
+    desc.add<std::string>("seedingMode", "Candidate");
+
+    desc.add<edm::InputTag>("input", edm::InputTag(""));
+    edm::ParameterSetDescription descPoints;
+    descPoints.add<std::vector<double>> ("eta", {} ); 
+    descPoints.add<std::vector<double>> ("phi", {} ); 
+    desc.add<edm::ParameterSetDescription>("points", descPoints);
+
+    desc.add<int>("maxNRegions", 10);
+    desc.add<edm::InputTag>("beamSpot", edm::InputTag("hltOnlineBeamSpot"));
+    desc.add<edm::InputTag>("vertexCollection", edm::InputTag("hltPixelVertices"));
+    desc.add<int>("maxNVertices", 1);
+
+    desc.add<double>("ptMin", 0.9);
+    desc.add<double>("originRadius", 0.2);
+    desc.add<double>("zErrorBeamSpot", 24.2);
+    desc.add<double>("deltaEta", 0.5);
```

Comment:
deltaEta and deltaPhi parameters are redundant, only included here to be able to reuse the very same configurations already in the menu. When switching to this combined producee, however, it is better to fill the correct parameters: in the interim period, HLT config parameters can be customized (as it has to be done anyhow because of the modified producers)

Reply:
Then, another possibility could be removing the "seedingMode" parameter from the configuration and define (e.g., just a possible suggestion)
if (deltaEta_Cand >=0 && deltaEta_Point>=0) m_seedingMode = SeedingMode::CANDIDATE_POINT_SEEDED;
else if (deltaEta_Cand >=0) m_seedingMode = SeedingMode::CANDIDATE_SEEDED;
else if (deltaEta_Point>=0) m_seedingMode = SeedingMode::POINT_SEEDED;
else throw (..)